content/How to work with Stripe.md

```diff
@@ -0,0 +1,349 @@
+  - `card_id` : Stripe card identifier
+
+The `token_id` you will get from your frontend is a short lived token that is only needed for an atomic operation
+
+Add a `customer_id` field to your user (or Shop in next example)
+Add a `card_id` to your user (or Shop in next example)
+
+Now take this service example (Shopify page example)
+``` ruby
+# app/services/stripe_service.rb
+
+require 'stripe'
+
+class StripeService
+  class StripeException < StandardError
+  end
+
+  attr_reader :shop
+
+  def initialize(shop)
+    @shop = shop
+  end
+
+  def add_card(token_id, email, name)
+    create_customer(email, name) unless customer.present?
+    card = customer.sources.create(source: token_id)
+    shop.update!(card_id: card.id)
+  end
+
+  def credit_card_info
+    card = shop.stripe_token
+    customer.sources.retrieve(card) if card
+  end
+
+  def update_credit_card(token_id)
+    card = customer.sources.create(source: token_id)
+    card.save
+    card_id = card.id
+    customer.default_source = card_id
+    customer.save
+    shop.update!(card_id: card_id)
+  end
+
+  def customer
+    customer_id = shop.customer_id
+    return unless customer_id.present?
+
+    @customer ||= Stripe::Customer.retrieve(customer_id)
+  end
+
+  private
+
+  def create_customer(email, name)
+    customer_params = {
+      email: email,
+      description: "#{shop.name} #{name}"
+    }
+    @customer = Stripe::Customer.create(customer_params)
+    shop.update!(customer_id: @customer.id)
+  end
+end
+```
+
+And this simple controller:
+
+```ruby
+# app/controllers/api/v1/credit_cards_controller.rb
+module Api
+  module V1
+    class CreditCardsController < Api::V1::ApiController
+      helper_method :shop
+
+      def index
+        service = StripeService.new(shop)
+        @card = service.credit_card_info
+        @customer = service.customer
+      end
+
+      def create
+        StripeService.new(shop).add_card(token_id, email, customer_name) if token_id
+        head :no_content
+      end
+
+      def update
+        begin
+          StripeService.new(shop).update_credit_card(token_id)
+        rescue StripeService::StripeException => ex
+          return render_not_found(ex.message)
+        end
+        head :no_content
+      end
+
+      private
+
+      def shop
+        @shop ||= current_shop
+      end
+
+      def token_json
+        params[:token]
+      end
+
+      def token_id
+        token_json['id']
+      end
+
+      def email
+        token_json['email']
+      end
+
+      def customer_name
+        token_json['name']
+      end
+    end
+  end
+end
+```
+
+And thats all! you can start charging your users now!
+All fraud detections and customer service actions can be managed directly from stripe dashboard.
+
+### SUBSCRIPTIONS
+To create a subscription you need to define them in your app and a product in stripe (this last one is really clear looking at dashboard, so i'm not going to explain it)
+
+##### CREATING THEM
```

Comment:
Maybe be more explicit here, instead of `THEM` say `CREATING THE/A SUBSCRIPTION`. Titles are easily seen if a reader takes a quick look at the article and they wouldn't understand what `THEM` means if they don't read it all.